### PR TITLE
Remove ObjC categories

### PR DESCRIPTION
**Pull request by bamx23**
There were several ObjC categories that weren't critical for the code but required linker ObjC flag (not really, but there was a hack to make it work).

Bye! Farewell!

### CODE REVIEW

File: Sources/KSCrashReportingCore/KSHTTPMultipartPostBody.m [hunk 1]
```diff
@@ -26,8 +26,21 @@
 
 #import "KSHTTPMultipartPostBody.h"
 
-#import "NSMutableData+AppendUTF8.h"
-#import "NSString+URLEncode.h"
+static void appendUTF8String(NSMutableData *data, NSString *string)
```
Comment (GLinnik21) [inline Sources/KSCrashReportingCore/KSHTTPMultipartPostBody.m:29]: It became less OOP)

File: Sources/KSCrashReportingCore/KSNSDictionaryHelper.m [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+static BOOL isNumericString(NSString *str)
+{
+    if ([str length] == 0) {
+        return YES;
+    }
+    unichar ch = [str characterAtIndex:0];
+    return ch >= '0' && ch <= '9';
+}
+
+static id objectForDeepKey(id container, NSArray *deepKey)
```
Comment (GLinnik21) [inline Sources/KSCrashReportingCore/KSNSDictionaryHelper.m:38]: Why not in Obj-C?

**Review by GLinnik21 — APPROVED** [review]
As we have removed all extensions, we no longer need the `-ObjC` flag in the compiler flags. Please consider removing this statement from the README.

File: Sources/KSCrashReportingCore/KSHTTPMultipartPostBody.m (hunk 1 above)
Comment (bamx23) [reply Sources/KSCrashReportingCore/KSHTTPMultipartPostBody.m:29]: Yeah, but at the same time categories are not really OOP :)

File: Sources/KSCrashReportingCore/KSNSDictionaryHelper.m (hunk 2 above)
Comment (bamx23) [reply Sources/KSCrashReportingCore/KSNSDictionaryHelper.m]: No reason, made ObjC

**Comment by bamx23** [comment]
Haven't found any `-ObjC` reference in the repo, by the way

**Pull request merged**